java/java.lsp.server/test/unit/src/org/netbeans/modules/java/lsp/server/ServerTest.java

```diff
@@ -0,0 +1,569 @@
+            }
+
+            @Override
+            public void logMessage(MessageParams arg0) {
+                throw new UnsupportedOperationException("Not supported yet.");
+            }
+        }, client.getInputStream(), client.getOutputStream());
+        serverLauncher.startListening();
+        LanguageServer server = serverLauncher.getRemoteProxy();
+        InitializeResult result = server.initialize(new InitializeParams()).get();
+        server.getTextDocumentService().didOpen(new DidOpenTextDocumentParams(new TextDocumentItem(src.toURI().toString(), "java", 0, code)));
+        Position pos = new Position(3, 30);
+        List<? extends Location> definition = server.getTextDocumentService().definition(new TextDocumentPositionParams(new TextDocumentIdentifier(src.toURI().toString()), pos)).get();
+        assertEquals(1, definition.size());
+        assertEquals(src.toURI().toString(), definition.get(0).getUri());
+        assertEquals(1, definition.get(0).getRange().getStart().getLine());
+        assertEquals(4, definition.get(0).getRange().getStart().getCharacter());
+        pos = new Position(4, 30);
+        definition = server.getTextDocumentService().definition(new TextDocumentPositionParams(new TextDocumentIdentifier(src.toURI().toString()), pos)).get();
+        assertEquals(1, definition.size());
+        assertEquals(src.toURI().toString(), definition.get(0).getUri());
+        assertEquals(2, definition.get(0).getRange().getStart().getLine());
+        assertEquals(23, definition.get(0).getRange().getStart().getCharacter());
+        //XXX: test jump to another file!
+    }
+
+    @Test
+    public void testOpenProjectOpenJDK() throws Exception {
+        getWorkDir().mkdirs();
+
+        FileObject root = FileUtil.toFileObject(getWorkDir());
+        try (Writer w = new OutputStreamWriter(FileUtil.createData(root, "jdk/src/java.base/share/classes/java/lang/Object.java").getOutputStream(), StandardCharsets.UTF_8)) {
+            w.write("package java.lang; public class Object {}");
+        }
+        FileUtil.createData(root, "jdk/src/java.base/share/classes/impl/Service.java");
+        FileObject javaBaseMI = FileUtil.createData(root, "jdk/src/java.base/share/classes/module-info.java");
+        try (Writer w = new OutputStreamWriter(javaBaseMI.getOutputStream(), StandardCharsets.UTF_8)) {
+            w.write("module java.base { exports java.lang; }");
+        }
+        try (Writer w = new OutputStreamWriter(FileUtil.createData(root, "jdk/src/java.compiler/share/classes/module-info.java").getOutputStream(), StandardCharsets.UTF_8)) {
+            w.write("module java.compiler { }");
+        }
+
+        ServerSocket srv = new ServerSocket(0);
+        Thread serverThread = new Thread(() -> {
+            try {
+                Socket server = srv.accept();
+                Server.run(server.getInputStream(), server.getOutputStream());
+            } catch (Exception ex) {
```

Comment:
Jackpot:
warning: Can be replaced with multicatch or several catch clauses catching specific exceptions